### PR TITLE
Feature/5947 tests contact journal extension/7375 extend tests

### PR DESCRIPTION
**Pull request by KaiTeuber**
## Description
* added a new launch argument to inject some high or low (positiv / negativ) corona test results to the contact journal
* also set that argument on the screen shot run
* unit tests checked, all have been updated befre

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-7375

## Screenshots
Simulator Screen Shot - Clone 1 of iPhone 12 Pro - 2021-05-26 at 11 49 52

### CODE REVIEW

**Review by ArturFriesen — APPROVED** [review]

File: src/xcode/ENA/ENAUITests/ContactJournal/ENAUITests_07_ContactJournalUITests.swift [hunk 1]
```diff
@@ -424,6 +425,31 @@ class ENAUITests_07_ContactJournal: XCTestCase {
...
+		app.setLaunchArgument(LaunchArguments.contactJournal.testsRiskLevel, to: "low")
+
+		navigateToJournalOverview()
+
+		// check if overview is visible
+		XCTAssertEqual(app.navigationBars.firstMatch.identifier, app.localized("ContactDiary_Overview_Title"))
+
+		// first cell should have the text for high risk, but none about checkin
+		let overviewCellWithEncounterRisk = app.descendants(matching: .table).firstMatch.cells.element(boundBy: 1)
+		let highRiskCell = overviewCellWithEncounterRisk.staticTexts[AccessibilityIdentifiers.ContactDiaryInformation.Overview.riskLevelLow]
```
Comment (pascal-brause) [inline src/xcode/ENA/ENAUITests/ContactJournal/ENAUITests_07_ContactJournalUITests.swift:442]:
```suggestion
		let lowRiskCell = overviewCellWithEncounterRisk.staticTexts[AccessibilityIdentifiers.ContactDiaryInformation.Overview.riskLevelLow]
```

File: src/xcode/ENA/ENAUITests/ContactJournal/ENAUITests_07_ContactJournalUITests.swift [hunk 2]
```diff
@@ -424,6 +425,31 @@ class ENAUITests_07_ContactJournal: XCTestCase {
...
+		XCTAssertTrue(highRiskCell.isHittable)
```
Comment (pascal-brause) [inline src/xcode/ENA/ENAUITests/ContactJournal/ENAUITests_07_ContactJournalUITests.swift:443]: Could you check if waitForExistence also works and replace it, when so? Also in the other UITest

File: src/xcode/ENA/ENAUITests/ContactJournal/ENAUITests_07_ContactJournalUITests.swift [hunk 3]
```diff
@@ -424,6 +425,31 @@ class ENAUITests_07_ContactJournal: XCTestCase {
...
+		let checkinCell = overviewCellWithEncounterRisk.staticTexts[AccessibilityIdentifiers.ContactDiaryInformation.Overview.checkinRiskLevelLow]
+		XCTAssertFalse(checkinCell.isHittable)
```
Comment (pascal-brause) [inline src/xcode/ENA/ENAUITests/ContactJournal/ENAUITests_07_ContactJournalUITests.swift:445]: same here

File: src/xcode/ENA/ENAUITests/ContactJournal/ENAUITests_07_ContactJournalUITests.swift [hunk 4]
```diff
@@ -424,6 +425,31 @@ class ENAUITests_07_ContactJournal: XCTestCase {
...
+
+		let overviewCellEmpty = app.descendants(matching: .table).firstMatch.cells.element(boundBy: 4)
+		let highRiskCellEmpty = overviewCellEmpty.staticTexts[AccessibilityIdentifiers.ContactDiaryInformation.Overview.riskLevelHigh]
+		XCTAssertFalse(highRiskCellEmpty.isHittable)
```
Comment (pascal-brause) [inline src/xcode/ENA/ENAUITests/ContactJournal/ENAUITests_07_ContactJournalUITests.swift:449]: same here

File: src/xcode/ENA/ENAUITests/ContactJournal/ENAUITests_07_ContactJournalUITests.swift [hunk 5]
```diff
@@ -424,6 +425,31 @@ class ENAUITests_07_ContactJournal: XCTestCase {
...
+		let checkinCellEmpty = overviewCellEmpty.staticTexts[AccessibilityIdentifiers.ContactDiaryInformation.Overview.checkinRiskLevelHigh]
+		XCTAssertFalse(checkinCellEmpty.isHittable)
```
Comment (pascal-brause) [inline src/xcode/ENA/ENAUITests/ContactJournal/ENAUITests_07_ContactJournalUITests.swift:451]: same here

**Review by pascal-brause — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/ContactDiaryStore.swift [hunk 6]
```diff
@@ -875,6 +875,10 @@ class ContactDiaryStore: DiaryStoring, DiaryProviding, SecureSQLStore {
 		return removeAllEntries(from: "ContactPerson")
 	}
 
+	func removeAllCoronaTests() -> Result<Void, SecureSQLStoreError> {
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/ContactDiaryStore.swift:878]: Could you add a unit test for that funktion, please.

Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/ContactDiaryStore.swift:878]: done

**Review by freshking — APPROVED** [review]

**Pull request merged**